wasm-runtime/polkadot/src/support/environment.rs

```diff
@@ -0,0 +1,33 @@
+use runtime_support::{Rc, RefCell, transmute, Box};
+use primitives::{BlockNumber, Digest};
+
+#[derive(Default)]
+pub struct Environment {
+	pub block_number: BlockNumber,
+	pub digest: Digest,
+	pub next_log_index: usize,
+}
+
+pub fn with_env<T, F: FnOnce(&mut Environment) -> T>(f: F) -> T {
+	let e = env();
+	let mut eb = e.borrow_mut();
+	f(&mut *eb)
+}
+
+pub fn env() -> Rc<RefCell<Environment>> {
```

Comment:
this whole function seems like it could be replaced by `lazy_static`

Reply:
i don't want to bring in non no-std crates until i know more about the restrictions we'll be under for wasm.